frontend/src/api/calls/studentCalls.ts

```diff
@@ -45,3 +54,18 @@ export function constructStudentQueryUrl(url: string, params: IStudentQueryParam
 
     return getQueryUrlFromParams(url, queryParams);
 }
+
+export async function getAllStudentInfo(studentUrl: string): Promise<IAllStudentInfo> {
+    const student: IStudent = await getStudentOnUrl(studentUrl);
+    const suggestions: ISuggestion[] = await getAllSuggestionsFromLinks(student._links.suggestions.href);
+    const fullSuggestions: IFullSuggestion[] = await Promise.all(
+        suggestions.map((suggestion: ISuggestion) => getFullSuggestionFromSuggestion(suggestion))
+    );
+    let skills: ISkillType[] = [];
+    for (let item of student.skills) {
+        const skill = await getSkillTypeWithName(item);
```

Comment:
Seems like a lot of api calls?

Reply:
Using hooks instead of call functions might resolve this problem. Maybe I need to explain this in a call?

Reply:
A student has at most 2 skills, so this is feasible, right?

Reply:
Sure, it's feasible but we have much more fitting ways to handle this.